src/Workspaces/Core/Portable/Serialization/SerializableSourceText.cs

```diff
@@ -139,13 +139,11 @@ public SourceText GetText(CancellationToken cancellationToken)
         }
     }
 
-    public void Serialize(ObjectWriter writer, SolutionReplicationContext context, CancellationToken cancellationToken)
+    public void Serialize(ObjectWriter writer, CancellationToken cancellationToken)
     {
         cancellationToken.ThrowIfCancellationRequested();
         if (_storage is not null)
         {
-            context.AddResource(_storage);
```

Comment:
the underlying _storage instance for a SerializableSourceText is never disposed.  So there's no way for hte memory it points to to go away.  So there's no need to root it in this context object.